tests/test_view/test_views.py

```diff
@@ -350,39 +163,56 @@ async def check_ws_response2(ws):
-    assert resp1.status_code == 200
-    assert not resp1.json()
+@Parametrization.autodetect_parameters()
+@Parametrization.case(name="non-exist-store-unauthorized", store_exists=True, authorized=False)
+@Parametrization.case(name="non-exist-store-authorized", store_exists=False, authorized=True)
+@Parametrization.case(name="store-unauthorized", store_exists=True, authorized=False)
+@Parametrization.case(name="store-authorized", store_exists=True, authorized=True)
+def test_ping_email(client: TestClient, token: str, store, store_exists, authorized):
+    store_id = store.id if store_exists else 999
+    resp = client.get(f"/stores/{store_id}/ping", headers={"Authorization": f"Bearer {token}"} if authorized else {})
+    if authorized:
+        if store_exists:
+            assert resp.status_code == 200
+            assert not resp.json()
+        else:
+            assert resp.status_code == 404
+    else:
+        assert resp.status_code == 401
 
 
-def test_user_stats(client: TestClient, token: str):
-    resp = client.get("/users/stats", headers={"Authorization": f"Bearer {token}"})
+@pytest.mark.asyncio
+async def test_user_stats(async_client, user: dict, token, store_wallet: dict):
+    store_id = store_wallet["store"].id
+    await create_product(user["id"], store_id=store_id)
+    await create_invoice(user["id"], store_id=store_id)
+    await create_invoice(user["id"], store_id=store_id)
+    resp = await async_client.get("/users/stats", headers={"Authorization": f"Bearer {token}"})
     assert resp.status_code == 200
     assert resp.json() == {
-        "wallets": 3,
-        "stores": 2,
-        "discounts": 1,
+        "discounts": 0,
+        "invoices": 2,
+        "notifications": 0,
         "products": 1,
-        "invoices": 0,
-        "notifications": 1,
-        "templates": 1,
+        "stores": 1,
+        "templates": 0,
+        "wallets": 1,
         "balance": 0.01,
     }
 
 
-def test_categories(client: TestClient):
-    assert client.get("/products/categories").status_code == 422
-    resp = client.get("/products/categories?store=1")
-    resp2 = client.get("/products/categories?store=2")
+@pytest.mark.asyncio
+@Parametrization.autodetect_parameters()
+@Parametrization.case(name="single", categories=["all"])
+@Parametrization.case(name="multiple", categories=["all", "test"])
+async def test_categories(async_client, user: dict, categories: list, store):
+    assert (await async_client.get("/products/categories")).status_code == 422
+    store_id = store.id
+    for category in categories:
+        await create_product(user["id"], store_id=store_id, category=category)
+    resp = await async_client.get(f"/products/categories?store={store_id}")
     assert resp.status_code == 200
-    assert resp.json() == ["all"]
-    assert resp2.status_code == 200
-    assert resp2.json() == ["all", "Test"]
+    assert resp.json() == categories
 
 
 def check_token(result):
```

Comment:
Note: it is now unused. Can either be removed or just the checking code can move here.

I think we should add a flake8 plugin to check for dead code automatically